src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/controllers/web/GitControllerTests.kt

```diff
@@ -27,4 +29,59 @@ class GitControllerTests : ControllerTest()
         val result = sut.getCommits()
         assertThat(result).isEqualTo(Serializer.instance.toResult(listOf(1, 2, 3)))
     }
+
+    @Test
+    fun `fetch receives successful fetch response from orderly, and returns git successful branches response`()
+    {   
+        val fetchResponse = OrderlyServerResponse(Serializer.instance.toResult("fetchResponse"), 200)
+        val branchesResponse = OrderlyServerResponse(Serializer.instance.toResult("branchesResponse"), 200)
+
+        val mockOrderlyServer = mock<OrderlyServerAPI>{
+            on { it.post("/v1/reports/git/fetch/", mockContext) } doReturn fetchResponse
+            on { it.get("/git/branches", mockContext) } doReturn branchesResponse
+        }
+
+        val sut = GitController(mockContext, mockOrderlyServer)
+        val response = sut.fetch()
+
+        assertThat(response).isEqualTo(Serializer.instance.toResult("branchesResponse"))
+    }
+
+    @Test
+    fun `fetch receives and returns failed fetch response from orderly, and does not call git branches`()
+    {   
+        val mockContext2 = mock<ActionContext>()
+        val fetchResponse = OrderlyServerResponse(Serializer.instance.toResult("fetchResponse"), 500)
+        val branchesResponse = OrderlyServerResponse(Serializer.instance.toResult("branchesResponse"), 200)
+
+        val mockOrderlyServer = mock<OrderlyServerAPI>{
+            on { it.post("/v1/reports/git/fetch/", mockContext2) } doReturn fetchResponse
+            on { it.get("/git/branches", mockContext2) } doReturn branchesResponse
+        }
+
+        val sut = GitController(mockContext2, mockOrderlyServer)
+        val response = sut.fetch()
+
+        assertThat(response).isEqualTo(Serializer.instance.toResult("fetchResponse"))
+        verify(mockContext2).setStatusCode(500)
```

Comment:
Suggest adding a new line to confirm that it `does not call git branches` (consistent with name of test):
```suggestion
        verify(mockContext2).setStatusCode(500)
        verify(mockOrderlyServer, times(0)).get(any(), any())
```